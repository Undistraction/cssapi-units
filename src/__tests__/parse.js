import {
  elementsOfUnitedNumber,
  numericPartOfUnitedNumber,
  unitPartOfUnitedNumber,
} from '../index';
import { genericInvalidValues, genericNumbers } from './data';

describe('elementsOfUnitedNumber', () => {
  it("throws if the value isn't a number followed by a unit", () => {
    for (const value of genericInvalidValues()) {
      expect(() =>
        elementsOfUnitedNumber(value)
      ).toThrowErrorMatchingSnapshot();
    }
  });

  it('returns the separate numeric and unit parts', () => {
    expect(elementsOfUnitedNumber('5px')).toEqual([5, 'px']);
    expect(elementsOfUnitedNumber('-10rem')).toEqual([-10, 'rem']);
    expect(elementsOfUnitedNumber('0em')).toEqual([0, 'em']);
    expect(elementsOfUnitedNumber('32%')).toEqual([32, '%']);
  });
});

describe('numericPartOfUnitedNumber', () => {
  it("throws if the supplied value isn't a united number", () => {
    const invalidValues = [...genericInvalidValues, ...genericNumbers];

    for (const value of invalidValues) {
      expect(() =>
        numericPartOfUnitedNumber(value)
      ).toThrowErrorMatchingSnapshot();
    }
  });

  it('returns the numeric part of a united number', () => {
    expect(numericPartOfUnitedNumber('5px')).toEqual(5);
    expect(numericPartOfUnitedNumber('-55em')).toEqual(-55);
    expect(numericPartOfUnitedNumber('33.6rem')).toEqual(33.6);
  });
});

describe('unitPartOfUnitedNumber', () => {
  it("throws if the supplied value isn't a united number", () => {
    const invalidValues = [
      ...genericInvalidValues,
      ...genericNumbers,
      'xxem',
      'NaNem',
    ];
    for (const value of invalidValues) {
      expect(() =>
        unitPartOfUnitedNumber(value)
      ).toThrowErrorMatchingSnapshot();
    }
  });

  it('returns the numeric part of a united number', () => {
    expect(unitPartOfUnitedNumber('5px')).toEqual('px');
    expect(unitPartOfUnitedNumber('-55em')).toEqual('em');
    expect(unitPartOfUnitedNumber('33.6rem')).toEqual('rem');
  });
});

// unitPartOfUnitedNumber
