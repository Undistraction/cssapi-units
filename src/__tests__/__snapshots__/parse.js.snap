// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 1`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: '5'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 2`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: '5'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 3`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'px'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 4`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'px10'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 5`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: '5 px'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 6`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: ''"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 7`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'null'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 8`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: '[object Object]'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 9`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'undefined'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 10`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: ''"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 11`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'true'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 12`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'false'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 13`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'NaN'"`;

exports[`elementsOfUnitedNumber throws if the value isn't a number followed by a unit 14`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: '0'"`;

exports[`unitPartOfUnitedNumber throws if the supplied value isn't a united number 1`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'xxem'"`;

exports[`unitPartOfUnitedNumber throws if the supplied value isn't a united number 2`] = `"[cssapi] elementsOfUnitedNumber() Supplied value was not a number with a unit: 'NaNem'"`;
